ros_buildfarm/templates/snippet/add_additional_repositories.Dockerfile.em

```diff
@@ -30,10 +33,11 @@ RUN @(' && '.join(commands))
 commands = []
 for component in ('contrib', 'non-free'):
     entry = 'deb http://httpredir.debian.org/debian %s %s' % (os_code_name, component)
-    commands.append('(grep -q -F -x -e "%s" /etc/apt/sources.list || echo "%s" >> /etc/apt/sources.list)' % (entry, entry))
+    pattern = 'deb http://httpredir\.debian\.org/debian/? %s ([-a-z] )*%s( [-a-z])*' % (os_code_name, component)
+    commands.append('(grep -q -E -x -e "%s" /etc/apt/sources.list || echo "%s" >> /etc/apt/sources.list)' % (pattern, entry))
 }@
 RUN @(' && '.join(commands))
 # Hit cloudfront mirror because of corrupted packages on fastly mirrors (https://github.com/ros-infrastructure/ros_buildfarm/issues/455)
 # You can remove this line to target the default mirror or replace this to use the mirror of your preference
-RUN sed -i 's/httpredir.debian.org/cloudfront.debian.net/' /etc/apt/sources.list
+RUN sed -i 's/httpredir\.debian\.org/cloudfront.debian.net/' /etc/apt/sources.list
```

Comment:
Is there a reason to only escape two of the dots?

Reply:
> Is there a reason to only escape two of the dots?

This is a sed substitution string so only the portion within the first // is treated as a regular expression. The text  after the second `/`, `cloudfront.debian.net` is the replacement string and is already a literal (if it included back-references like `\1` it wouldn't be truly a literal, but the '.' characters would still be literal).

Reply:
Oh, I see. Too much different tools going on :wink: